crypto/ec/ecx_meth.c

```diff
@@ -16,30 +16,88 @@
 #include "internal/evp_int.h"
 #include "ec_lcl.h"
 
-#define X25519_KEYLEN        32
 #define X25519_BITS          253
 #define X25519_SECURITY_BITS 128
 
 #define ED25519_SIGSIZE      64
 
-typedef struct {
-    unsigned char pubkey[X25519_KEYLEN];
-    unsigned char *privkey;
-} X25519_KEY;
+#define X448_BITS            448
+#define ED448_BITS           456
+#define X448_SECURITY_BITS   224
+
+#define ED448_SIGSIZE        114
+
+#define ISX448(id)      ((id) == EVP_PKEY_X448 ? 1 : 0)
+#define IS25519(id)     (((id) == EVP_PKEY_X25519 || (id) == EVP_PKEY_ED25519) \
+                            ? 1 : 0)
+#define KEYLENID(id)    (IS25519(id) ? X25519_KEYLEN \
+                                     : ((id) == EVP_PKEY_X448 ? X448_KEYLEN \
+                                                              : ED448_KEYLEN))
+#define KEYLEN(p)       KEYLENID((p)->ameth->pkey_id)
+
+#define MAX_KEYLEN  ED448_KEYLEN
+
+#define GETPRIVKEY(priv, pk) \
```

Comment:
Any reason why you prefer a define instead of a static (ossl_inline) function?

Reply:
No, not really. Probably an inline function would be more readable, so I made that change.